File: src/server/go/srv.go
package main

import (
	"github.com/gin-gonic/gin"
	"net/http"
)

type Version struct {
	VersionId   string `json:"versionId"`
	Version     string `json:"version"`
	ReleaseDate string `json:"releaseDate"`
	DownloadUrl string `json:"downloadUrl"`
}

var currentVersion = Version{
	VersionId:   "1",
	Version:     "1.2.3",
	ReleaseDate: "2023-07-15",
	DownloadUrl: "https://example.com/downloads/1.2.3/app.zip",
}

func main() {
	router := gin.Default()

	router.GET("/version", func(c *gin.Context) {
		c.JSON(http.StatusOK, currentVersion)
	})

	router.GET("/download/:versionId", func(c *gin.Context) {
		requestedVersionId := c.Param("versionId")
		if requestedVersionId == currentVersion.VersionId {
			c.Redirect(http.StatusTemporaryRedirect, currentVersion.DownloadUrl)
		} else {
			c.JSON(http.StatusNotFound, gin.H{"error": "Requested version not found"})
		}
	})

	port := ":3000"
	router.Run(port)
}

File: src/server/php/srv.php
<?php

$versions = [
    [
        "versionId" => "1",
        "version" => "1.2.3",
        "releaseDate" => "2023-07-15",
        "downloadUrl" => "https://example.com/downloads/1.2.3/app.zip"
    ]
];

// Sürüm bilgisini getiren endpoint
if ($_SERVER['REQUEST_METHOD'] === 'GET' && isset($_GET['action']) && $_GET['action'] === 'version') {
    echo json_encode($versions[0]);
    exit;
}

// İndirme linkini kontrol eden endpoint
if ($_SERVER['REQUEST_METHOD'] === 'GET' && isset($_GET['action']) && $_GET['action'] === 'download') {
    if (isset($_GET['versionId'])) {
        $requestedVersionId = $_GET['versionId'];
        foreach ($versions as $version) {
            if ($version['versionId'] === $requestedVersionId) {
                header("Location: " . $version['downloadUrl']);
                exit;
            }
        }
    }

    // Eğer belirtilen sürüm bulunamazsa 404 hatası döndürür
    http_response_code(404);
    echo json_encode(["error" => "Requested version not found."]);
    exit;
}

// Diğer durumlarda 404 hatası döndürür
http_response_code(404);
echo json_encode(["error" => "Not Found"]);
exit;
